java/src/shared/IServer.java

```diff
@@ -108,7 +116,7 @@
      * @post If the operation succeeds,  1. The server returns an HTTP 200 success response.  2. The body contains a JSON array of  commands that have been executed in the game.  This command array is suitable for passing back to the /game/command [POST] method to  restore the state of the game later (after calling /game/reset to revert the game to its initial state).     If the operation fails,  1. The server returns an HTTP 400 error response, and the body contains an error message.
      * @return true if the list of commands is returned; false otherwise
      */
-    boolean getCommandsGame();
+    Gson getCommandsGame();
```

Comment:
`IServer` should not be returning `Gson` objects. Gson is _only_ used for sending to/from the server and parameters will only take it in the `ClientCommunicator`.